File: java/app/src/main/java/com/youknow/movie/ui/upcoming/UpcomingContract.java
package com.youknow.movie.ui.upcoming;

import com.youknow.domain.model.SimpleMovie;

import java.util.List;

public interface UpcomingContract {
    interface View {
        void showProgressBar(int visibility);
        void onMoviesLoaded(List<SimpleMovie> movies);
        void onError(int msgResId);
        void hideError();
    }

    interface Presenter {
        void getMoviesNowPlaying();
    }
}

File: kotlin-rx-koin/app/src/main/java/com/youknow/movie/ui/upcoming/UpcomingContract.kt
package com.youknow.movie.ui.upcoming

import com.youknow.domain.model.SimpleMovie

interface UpcomingContract {
    interface View {
        fun showProgressBar(visible: Int)
        fun hideError()
        fun onError(msgResId: Int)
        fun onMoviesLoaded(movies: List<SimpleMovie>)
    }

    interface Presenter {
        fun unsubscribe()
        fun getUpcomingMovies()
    }
}